Examples/c/class/example.h

```diff
@@ -0,0 +1,39 @@
+/* File : example.h */
+
+class Shape {
+public:
+  Shape() {
+    nshapes++;
+  }
+  virtual ~Shape() {
+    nshapes--;
+  };
+  double  x, y;   
+  void    move(double dx, double dy);
+  virtual double area(void) = 0;
+  virtual double perimeter(void) = 0;
+  static  int nshapes;
+};
+
+class Circle : public Shape {
+private:
+  double radius;
+public:
+  Circle(double r) : radius(r) { };
+  virtual double area(void);
+  virtual double perimeter(void);
+};
+
+class Square : public Shape {
+private:
+  double width;
+public:
+  Square(double w) : width(w) { };
+  virtual double area(void);
+  virtual double perimeter(void);
+};
+
+
+
+
+  
```

Comment:
This looks like a copy of an old version of this example from another target language.  7+ years ago I cleaned these up to be consistent between languages (b115c984a97e5f20e9ad5979f1ad606fff549bc3 for example, but there are probably other commits) but here I notice at least `void` parameter lists and a superfluous `;` after `Shape`'s destructor's definition are back.

I realise this is due to how long-lived this branch has been, but it'd be good to compare the C examples with the equivalents from other target languages and avoid reintroducing these ancient inconsistencies.

Reply:
I've removed gratuitous differences between this file and the C# version, thanks.

Reply:
This is a wider issue - e.g. in master:

```
$ git grep 'Circle::area' Examples/*/class/example.cxx
Examples/csharp/class/example.cxx:double Circle::area() {
Examples/d/class/example.cxx:double Circle::area() {
Examples/guile/class/example.cxx:double Circle::area() {
Examples/java/class/example.cxx:double Circle::area() {
Examples/javascript/class/example.cxx:double Circle::area() {
Examples/lua/class/example.cxx:double Circle::area() {
Examples/mzscheme/class/example.cxx:double Circle::area() {
Examples/octave/class/example.cxx:double Circle::area() {
Examples/perl5/class/example.cxx:double Circle::area() {
Examples/php/class/example.cxx:double Circle::area() {
Examples/python/class/example.cxx:double Circle::area() {
Examples/r/class/example.cxx:double Circle::area() {
Examples/ruby/class/example.cxx:double Circle::area() {
Examples/scilab/class/example.cxx:double Circle::area() {
Examples/tcl/class/example.cxx:double Circle::area() {
```

But the version added for C here has `void` here (and there are also other long-fixed niggles reintroduced, such as the comment at the top saying the file is `example.c` when it's actually `example.cxx`).

I was suggesting comparing **all** the C examples with those from other target languages (you can probably just pick one that's in a reasonable state) e.g. something like: `diff -ru --exclude=Makefile --exclude=check.list Examples/csharp Example/c` to eliminate these sort of inconsistencies.

Reply:
I could have missed something because it wasn't simple to see the relevant differences among all the other ones, but I think I've got rid of all such problems now (see 4748e4f3798188c07bcd7b3a0a9ab159390d5169).